src/workflows-app/utils/method-common.js

```diff
@@ -14,7 +16,7 @@ const MethodSource = Object.freeze({
 const Covid19Methods = ['fetch_sra_to_bam', 'assemble_refbased', 'sarscov2_nextstrain'];
 export const isCovid19Method = (methodName) => Covid19Methods.includes(methodName);
 
-export const submitMethod = async (signal, onDismiss, method, workspace) => {
+export const submitMethod = async (signal, onDismiss, method, workspace, setImportLoading, setMethodId, setSuccessfulImport, setErrorMessage) => {
```

Comment:
Ah, source of weirdness located. Could we abstract out to "onSuccess" and "onError"? This would remove the load from `submitMethod` to know how to act in the two current use cases. Or we could just return the response from the Ajax call and make the caller of `submitMethod` handle the try/catch.